src/pages/patientView/sampleManager.tsx

```diff
@@ -18,10 +18,12 @@ export function placeArrow(tooltipEl: any) {
 class SampleManager {
 
     sampleIndex: { [s:string]:number };
+    sampleIdsInOrder: string[];
 
     constructor(public samples: Array<ClinicalDataBySampleId>){
 
         this.sampleIndex = {};
+        this.sampleIdsInOrder = samples.map((sample:ClinicalDataBySampleId) => sample.id);
```

Comment:
troubles me slightly that we now have two orderings:  1. the order of samples in samples collection and two the order in this array.  Is there a reason not to just iterate through samples collection in your calling code, using sample.id?  (I admit's it's a tiny bit weird that the samples collection is filled with things of type ClinicalDataBySampleId ... )